app/services/search_service/exact_search.rb

```diff
@@ -80,16 +80,27 @@ def find_by_chemical(query)
       matchdata = /\A(cas\s*)?(\d+-\d+-\d)\z/i.match(query)
       q = matchdata ? matchdata[2] : query.gsub(/\Acas\s+/i, '')
 
-      if c = Chemical.first(cas: q)
+      if (c = Chemical.first(cas: q))
         gns = c.goods_nomenclatures.map do |gn|
           ExactSearch.new(gn.goods_nomenclature_item_id, date).search!.results
         end
 
-        # Each Chemical should map to only one Goods Nomenclaure,
+        # Each Chemical should map to only one Goods Nomenclaure,
         # but the database includes two chemicals that belong to more than one GN
         # These "chemicals" are probably placeholders and are not really correct
         return gns.first if gns.length == 1
       end
     end
+
+    # Example:
+    # 'cookie' => ['cookie', cookies]
+    # 'leaves' => ['leaf', 'leaves']
+    def singular_and_plural(query)
+      if query == query.pluralize
+        [query.singularize, query]
```

Comment:
Is it simpler just to do:

```ruby
[
  query.singularize, # singular form
  query.pluralize,  # plural form
  query  # original form for edge cases
]
```

Reply:
I like it! and we can add `.uniq` to avoid duplication